src/dbus_fast/message_bus.py

```diff
@@ -945,24 +932,18 @@ def _find_message_handler(
             ):
                 return self._default_get_managed_objects_handler
 
-        msg_path = msg.path
-        if msg_path:
-            interfaces = self._path_exports.get(msg_path)
-            if not interfaces:
-                return None
-            for interface in interfaces:
-                methods = ServiceInterface._c_get_methods(interface)
-                for method in methods:
-                    if method.disabled:
-                        continue
-
-                    if (
-                        msg.interface == interface.name
-                        and msg.member == method.name
-                        and msg.signature == method.in_signature
-                    ):
-                        return ServiceInterface._c_get_handler(interface, method, self)
-
+        if (
+            msg.path is not None
+            and (interfaces := self._path_exports.get(msg.path)) is not None
+            and (interface := interfaces.get(msg.interface)) is not None
+            and (
+                handler := ServiceInterface._get_enabled_handler_by_name_signature(
+                    interface, self, msg.member, msg.signature
+                )
+            )
+            is not None
+        ):
+            return handler
```

Comment:
All of the refactoring was so we could do this .. which should have a time complexity of O(3) instead of O(x)*O(y)*O(z)